### DISCUSSION TITLE
[vite-plugin-federation] Consume federated modules from Webpack 4?

### DISCUSSION THREAD

**Discussion by austinknight** [originjs/origin.js#141] [Q&A] [answered]
Anyone done something like a federated module exported from [vite-plugin-federation](https://github.com/originjs/vite-plugin-federation) and consumed it in a webpack 4 app? I know in WP4 it does not support module federation natively so you would have to import a script at runtime, i've seen this done when federating with webpack 5. To clarify Host = webpack4 app and Remote = vite with vite-plugin-federation. Is this even possible with this vite plugin since the output is`esm`?

**Comment by austinknight** [comment, marked as answer]
I figured it out, just need to change the rollup output format to `es` then you can async load the `remoteEntry.js` and instantiate a module from the returned code.

**Reply by chitralverma** [reply]
Hi, can you please post and example to do this. 
Thanks!
